dominion-pax-exam-options/src/main/java/org/codice/dominion/pax/exam/options/KarafUsersPropertiesFileGroupPutOption.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.dominion.pax.exam.options;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.codice.dominion.options.karaf.KarafOptions;
+import org.ops4j.pax.exam.karaf.options.KarafDistributionConfigurationFilePutOption;
+
+/** PaxExam option for adding a new user or replacing an existing one. */
+public class KarafUsersPropertiesFileGroupPutOption
+    extends KarafDistributionConfigurationFilePutOption {
+  public static final String GROUP_PREFIX = "_g_:";
+
+  private final Set<String> roles;
+
+  /**
+   * Creates a new option.
+   *
+   * @param groupId the group id to be added or replaced
+   */
+  public KarafUsersPropertiesFileGroupPutOption(String groupId) {
+    super(KarafOptions.USER_PROPERTIES, groupId, null);
+    this.roles = new LinkedHashSet<>();
```

Comment:
:question: Does order matter for these  User Property classes?

Reply:
not sure. It does keep things consistent with how they are defined which makes it easier to follow